### PR TITLE
Add a runtime hoster crate that makes constructing a host environment around a runtime super easy

### PR DESCRIPTION
**Pull request by expenses**
Sort-of a start on https://github.com/paritytech/substrate/issues/8974. So far the `run-node` demo works but the `run-node-template` does not.

### CODE REVIEW

File: bin/hoster/src/lib.rs [hunk 1]
```diff
@@ -0,0 +1,397 @@
...
+	Extension:
+		sp_runtime::DeserializeOwned + Send + Sync + sc_service::ChainSpecExtension + 'static,
+{
+	fn impl_name() -> String {
+		"Runtime Hoster".into()
+	}
+
+	fn impl_version() -> String {
+		Default::default()
+	}
```
Comment (gilescope) [inline bin/hoster/src/lib.rs:324]: Do we want impl_version to be an empty string?

**Review by gilescope — APPROVED** [review]

File: bin/hoster/src/lib.rs (hunk 1 above)
Comment (expenses) [reply bin/hoster/src/lib.rs:37]: Probably not, but it'll do for now.

**Comment by expenses** [comment]
I'm not sure what options we want this builder to have, nor what runtime apis we should require.

**Comment by expenses** [comment]
No longer working on this. Feel free to take over.

**Pull request closed without merging**